syntax = "proto3";

package tests.harness;
option go_package = "harness";

import "google/protobuf/any.proto";

enum TestType {
    TestTypeValidate = 0;
    TestTypeAllErrors = 1;
}

message TestCase {
    google.protobuf.Any message = 1;
    TestType test_type = 2;
}

message TestResult {
    bool   Valid  = 1;
    bool   Error  = 2;
    string Reason = 3;
    bool   AllowFailure = 4;
    int32 ErrorCount = 5;
}
